### ISSUE TITLE
Reggie! Next - Recursion error when opening stage folder

### ISSUE THREAD

**Issue by CreeperMario** [RoadrunnerWMC/Reggie-Next#8]
Hello, RoadRunnerWMC or anyone who's reading this.
I have been using Reggie! for a while, and stumbled on this.
However, I cannot use it because it won't accept my stage folder.
The folder is from the original <i>New SUPER MARIO BROS. Wii</i> game disc, dumped using Wiimm's ISO Tools. The original Reggie! accepts the folder just fine (and loads every file from it), but when I try to load the same folder using Reggie! Next, it does this? (see http://pastebin.com/AMGPWiCN)
Although I have almost no knowledge of Python, I can see that the same functions (InitFromName, FindGameDef and ReggieGameDefinition) are being called over and over again, overloading Python's function stack.
I know that you're probably busy with other projects, but this is an issue that prevents me from using Reggie! Next.
Specs: Ubuntu 14.04.2 LTS, Python 3.4, PyQt (latest version available from apt-get) and TPLLib and PyQtRibbon downloaded from your repos. I am also using Reggie! Next Milestone 2 Alpha 3.
Cheers, CreeperMario

**Comment by Meorge** [comment]
A recursion error means it kept repeating itself a ton of times, and then finally decided it was in an infinite loop, so it stopped itself (note the large number of repeating errors). I don't know a lot about how game patches work, but when I get home I'll look into it.

-Meorge

**Comment by RoadrunnerWMC** [comment]
Hi, and thanks for trying out Reggie Next!

I've never had this particular problem before, but you're correct in your analysis of it. This isn't related to selecting the `Stage` folder, though; this is occurring after the `Stage` folder's contents have already been verified and accepted.

This bug is related to loading the game patches, somehow. Specifically, it looks like the `if` statement

    if node.attrib['game'] != trans.string('Gamedefs', 13): # 'New Super Mario Bros. Wii'

is firing when it shouldn't be.

Did you make any changes to anything in Reggie Next's `reggiedata/games/` folder?

**Comment by CreeperMario** [comment]
Hi, RoadrunnerWMC.
Unless my archiving program incorrectly extracted the files, no
modifications have been made to the folder. When I can, I will try
re-downloading the software (Milestone 2 Alpha 3 from the Releases section)
and if the error persists, I'll write back.
Once again, I almost have no knowledge of Python, but I'm good at Java (not
helping).

**Comment by CreeperMario** [comment]
Nope. A fresh download does not fix the program. Although, I have noticed, that inside <code>reggiedata/games</code> there are the folders <code>AnotherSMBW</code>, <code>NewerSMBW</code>, <code>NewerSMBWHS</code> and <code>NewerSumSun</code>, but no folder for <i>New SUPER MARIO BROS. WIi.</i> Is it supposed to be like that?

**Comment by RoadrunnerWMC** [comment]
If the `reggiedata/games` folder is no longer in the Reggie Next repo for some reason, then I can't really fix this bug -- it doesn't exist anymore in the repository version. The repository version should work for you, then.

I did look at the code for M2A3, and here's what I think is happening. Some game patches are able to reference others in various ways, to avoid duplicating files. When loading a game patch (the `__init__` and `InitFromName` functions), sometimes it will need to search for other game patches for that reason. The function that does the search (`FindGameDef`) needs to actually load each game patch in the `reggiedata/games` folder in order to find the right one. That function is supposed to skip the game patch that called it (with the `skip` argument). It appears that, on your computer, it isn't doing that properly, and it loads the same game patch again during its search. Loading that will cause another search to be performed, which will cause the same game patch to be loaded again, etc.

The main difference between your computer configuration (thank you for providing that information, by the way) and mine is that you're running Ubuntu, and I use Windows. I wonder if `os.listdir('reggiedata/games')` within `FindGameDef`, which lists all of the folders in a directory, behaves differently somehow on Ubuntu. That would influence the `folder` variable in the loop, which might make `if folder == skip: continue` not fire when it should. That would result in this infinite recursion error. That's one possible explanation, at least.

I can't reproduce this on my computer, and this doesn't really affect the repo version, either, so I'll close this issue now. If you debug it (I'd start by checking if `if folder == skip: continue` ever fires like it should), you can post your findings here and I'll re-open it. If you don't want to, that's fine, too -- I've already started [a complete rewrite of Reggie Next] (http://github.com/RoadrunnerWMC/Reggie-Next-2), anyway.

Either way, thanks for reporting this issue.

**Comment by CreeperMario** [comment]
Hello, developers!
So, an update on my situation (yes, it has been months since my last presence!)
I have scrapped Ubuntu, and now have a Mac dual-booting Windows 10.
I have tried running Reggie! Next underneath Windows, and it does indeed work out-of-the-box!

So, what I have learned today: Linux doesn't solve all of the world's problems.

Thank you so much for your work on the software, it is a very helpful tool for creating levels, great work!
-CreeperMario
